src/core/namespace.c

```diff
@@ -1305,6 +1359,79 @@ int bind_mount_add(BindMount **b, unsigned *n, const BindMount *item) {
         return 0;
 }
 
+void temporary_filesystem_free_many(TemporaryFileSystem *t, unsigned n) {
+        unsigned i;
+
+        assert(t || n == 0);
+
+        for (i = 0; i < n; i++) {
+                free(t[i].path);
+                free(t[i].options);
+        }
+
+        free(t);
+}
+
+int temporary_filesystem_add(
+                TemporaryFileSystem **t,
+                unsigned *n,
+                bool ignore_enoent,
+                const char *path,
+                const char *options) {
+
+        _cleanup_free_ char *p = NULL, *o = NULL, *flag_string = NULL;
+        unsigned long flags = MS_NODEV|MS_STRICTATIME;
+        TemporaryFileSystem *c;
+        bool read_only;
+        char *a;
+        int r;
+
+        assert(t);
+        assert(n);
+        assert(path);
+
+        p = strdup(path);
+        if (!p)
+                return -ENOMEM;
+
+        r = mnt_optstr_get_flags(options, &flags, mnt_get_builtin_optmap(MNT_LINUX_MAP));
+        if (r < 0)
+                return r;
+
+        r = mnt_optstr_apply_flags(&flag_string, flags, mnt_get_builtin_optmap(MNT_LINUX_MAP));
+        if (r < 0)
+                return r;
+
+        a = flag_string;
+        while((a = strchr(a, ',')))
+                *a = '\0';
```

Comment:
what is this loop supposed to do? i mean it looks for the first comma, patches it into an empty string and then looks in that empty string for more commands, which it's obviously not going to find... not sure i follow

Reply:
Indeed! Thanks.